arch/arm64/core/mmu.c

```diff
@@ -816,7 +816,12 @@ static void enable_mmu_el1(struct arm_mmu_ptables *ptables, unsigned int flags)
 	barrier_isync_fence_full();
 
 	/* Invalidate all data caches before enable them */
+#ifdef CONFIG_DT_HAS_XEN_XEN_ENABLED
```

Comment:
So there are a few issues with using set/way instructions in Zephyr:
  * As you said above they are not easily virtualizable
  * On Armv8, sytem caches are can only be cleaned by VA. So this would not flush all the caches.

If you want more details, there is a pretty good talk from Marc Zyngier about it:
https://www.youtube.com/watch?v=A_zCxzpxzmE

Therefore, I think whatever solution you come up with should be generic.

Reply:
Thank you for addition, we currently have discussion #58436 for this issue. We tried U-boot approach, but it will have issues during dynamic memory mapping. For sure, final solution should be generic, this commit is here only for test purposes during review.

Reply:
> Therefore, I think whatever solution you come up with should be generic.

Here is the proper fix instead of the current workaround: https://github.com/zephyrproject-rtos/zephyr/pull/61035